GC tests fail on PyPy3.11 7.3.19

Reproduced with fc0599e122edcde0f84962b219ea591428124106.

```pytb
$ tox -e pypy311-test
pypy311-test: install_deps> python -I -m pip install setuptools
.pkg-pypy311: _optional_hooks> python /usr/lib/python3.14/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg-pypy311: get_requires_for_build_editable> python /usr/lib/python3.14/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
.pkg-pypy311: build_editable> python /usr/lib/python3.14/site-packages/pyproject_api/_backend.py True setuptools.build_meta __legacy__
pypy311-test: install_package_deps> python -I -m pip install 'mypy>=1.14.0' pytest pytest-asyncio 'typing_extensions>=4; python_version < "3.11"'
pypy311-test: install_package> python -I -m pip install --force-reinstall --no-deps /tmp/asgiref/.tox/.tmp/package/2/asgiref-3.9.0-0.editable-py3-none-any.whl
pypy311-test: commands[0]> pytest -v
========================================================= test session starts =========================================================
platform linux -- Python 3.11.11[pypy-7.3.19-final], pytest-8.4.1, pluggy-1.6.0 -- /tmp/asgiref/.tox/pypy311-test/bin/python
cachedir: .tox/pypy311-test/.pytest_cache
rootdir: /tmp/asgiref
configfile: setup.cfg
testpaths: tests
plugins: asyncio-1.0.0
asyncio: mode=Mode.STRICT, asyncio_default_fixture_loop_scope=function, asyncio_default_test_loop_scope=function
collected 82 items                                                                                                                    

tests/test_compatibility.py::test_is_double_callable PASSED                                                                     [  1%]
tests/test_compatibility.py::test_double_to_single_signature PASSED                                                             [  2%]
tests/test_compatibility.py::test_double_to_single_communicator PASSED                                                          [  3%]
tests/test_garbage_collection.py::test_thread_critical_Local_remove_all_reference_cycles FAILED                                 [  4%]
tests/test_local.py::test_local_task PASSED                                                                                     [  6%]
tests/test_local.py::test_local_thread PASSED                                                                                   [  7%]
tests/test_local.py::test_local_thread_nested PASSED                                                                            [  8%]
tests/test_local.py::test_local_cycle PASSED                                                                                    [  9%]
tests/test_local.py::test_local_task_to_sync PASSED                                                                             [ 10%]
tests/test_local.py::test_local_thread_to_async PASSED                                                                          [ 12%]
tests/test_local.py::test_local_task_to_sync_to_task PASSED                                                                     [ 13%]
tests/test_local.py::test_local_many_layers PASSED                                                                              [ 14%]
tests/test_local.py::test_local_critical_no_task_to_thread PASSED                                                               [ 15%]
tests/test_local.py::test_local_critical_no_thread_to_task PASSED                                                               [ 17%]
tests/test_local.py::test_local_threads_and_tasks PASSED                                                                        [ 18%]
tests/test_local.py::test_thread_critical_local_not_context_dependent_in_sync_thread PASSED                                     [ 19%]
tests/test_local.py::test_visibility_thread_asgiref PASSED                                                                      [ 20%]
tests/test_local.py::test_visibility_task PASSED                                                                                [ 21%]
tests/test_server.py::test_stateless_server PASSED                                                                              [ 23%]
tests/test_server.py::test_server_delete_instance PASSED                                                                        [ 24%]
tests/test_sync.py::test_sync_to_async PASSED                                                                                   [ 25%]
tests/test_sync.py::test_sync_to_async_fail_non_function PASSED                                                                 [ 26%]
tests/test_sync.py::test_sync_to_async_fail_async PASSED                                                                        [ 28%]
tests/test_sync.py::test_async_to_sync_fail_partial PASSED                                                                      [ 29%]
tests/test_sync.py::test_sync_to_async_raises_typeerror_for_async_callable_instance PASSED                                      [ 30%]
tests/test_sync.py::test_sync_to_async_decorator PASSED                                                                         [ 31%]
tests/test_sync.py::test_nested_sync_to_async_retains_wrapped_function_attributes PASSED                                        [ 32%]
tests/test_sync.py::test_sync_to_async_method_decorator PASSED                                                                  [ 34%]
tests/test_sync.py::test_sync_to_async_method_self_attribute PASSED                                                             [ 35%]
tests/test_sync.py::test_async_to_sync_to_async PASSED                                                                          [ 36%]
tests/test_sync.py::test_async_to_sync_to_async_decorator PASSED                                                                [ 37%]
tests/test_sync.py::test_async_to_sync_to_thread_decorator PASSED                                                               [ 39%]
tests/test_sync.py::test_async_to_sync_fail_non_function PASSED                                                                 [ 40%]
tests/test_sync.py::test_async_to_sync_fail_sync PASSED                                                                         [ 41%]
tests/test_sync.py::test_async_to_sync PASSED                                                                                   [ 42%]
tests/test_sync.py::test_async_to_sync_decorator PASSED                                                                         [ 43%]
tests/test_sync.py::test_async_to_sync_method_decorator PASSED                                                                  [ 45%]
tests/test_sync.py::test_async_to_sync_in_async PASSED                                                                          [ 46%]
tests/test_sync.py::test_async_to_sync_in_thread PASSED                                                                         [ 47%]
tests/test_sync.py::test_async_to_sync_in_except PASSED                                                                         [ 48%]
tests/test_sync.py::test_async_to_sync_partial PASSED                                                                           [ 50%]
tests/test_sync.py::test_async_to_sync_on_callable_object PASSED                                                                [ 51%]
tests/test_sync.py::test_async_to_sync_method_self_attribute PASSED                                                             [ 52%]
tests/test_sync.py::test_thread_sensitive_outside_sync PASSED                                                                   [ 53%]
tests/test_sync.py::test_thread_sensitive_outside_async PASSED                                                                  [ 54%]
tests/test_sync.py::test_thread_sensitive_with_context_matches PASSED                                                           [ 56%]
tests/test_sync.py::test_thread_sensitive_nested_context PASSED                                                                 [ 57%]
tests/test_sync.py::test_thread_sensitive_context_without_sync_work PASSED                                                      [ 58%]
tests/test_sync.py::test_thread_sensitive_double_nested_sync PASSED                                                             [ 59%]
tests/test_sync.py::test_thread_sensitive_double_nested_async PASSED                                                            [ 60%]
tests/test_sync.py::test_thread_sensitive_disabled PASSED                                                                       [ 62%]
tests/test_sync.py::ASGITest::test_wrapped_case_is_collected PASSED                                                             [ 63%]
tests/test_sync.py::test_sync_to_async_detected_as_coroutinefunction PASSED                                                     [ 64%]
tests/test_sync.py::test_multiprocessing PASSED                                                                                 [ 65%]
tests/test_sync.py::test_sync_to_async_uses_executor PASSED                                                                     [ 67%]
tests/test_sync.py::test_sync_to_async_deadlock_ignored_with_exception PASSED                                                   [ 68%]
tests/test_sync.py::test_sync_to_async_with_blocker_thread_sensitive XFAIL                                                      [ 69%]
tests/test_sync.py::test_sync_to_async_with_blocker_non_thread_sensitive PASSED                                                 [ 70%]
tests/test_sync.py::test_sync_to_async_within_create_task PASSED                                                                [ 71%]
tests/test_sync.py::test_inner_shield_sync_middleware PASSED                                                                    [ 73%]
tests/test_sync.py::test_inner_shield_async_middleware PASSED                                                                   [ 74%]
tests/test_sync.py::test_inner_shield_sync_and_async_middleware PASSED                                                          [ 75%]
tests/test_sync.py::test_inner_shield_sync_and_async_middleware_sync_task PASSED                                                [ 76%]
tests/test_sync.py::test_async_to_sync_overlapping_kwargs PASSED                                                                [ 78%]
tests/test_sync.py::test_sync_to_async_overlapping_kwargs PASSED                                                                [ 79%]
tests/test_sync.py::test_nested_task PASSED                                                                                     [ 80%]
tests/test_sync.py::test_nested_task_later PASSED                                                                               [ 81%]
tests/test_sync.py::test_double_nested_task PASSED                                                                              [ 82%]
tests/test_sync.py::test_two_nested_tasks_with_asyncio_run PASSED                                                               [ 84%]
tests/test_sync_contextvars.py::test_thread_sensitive_with_context_different PASSED                                             [ 85%]
tests/test_sync_contextvars.py::test_sync_to_async_contextvars PASSED                                                           [ 86%]
tests/test_sync_contextvars.py::test_async_to_sync_contextvars PASSED                                                           [ 87%]
tests/test_testing.py::test_receive_nothing PASSED                                                                              [ 89%]
tests/test_testing.py::test_receive_nothing_lazy_loop PASSED                                                                    [ 90%]
tests/test_wsgi.py::test_basic_wsgi PASSED                                                                                      [ 91%]
tests/test_wsgi.py::test_script_name PASSED                                                                                     [ 92%]
tests/test_wsgi.py::test_wsgi_path_encoding PASSED                                                                              [ 93%]
tests/test_wsgi.py::test_wsgi_empty_body PASSED                                                                                 [ 95%]
tests/test_wsgi.py::test_wsgi_clamped_body PASSED                                                                               [ 96%]
tests/test_wsgi.py::test_wsgi_stops_iterating_after_content_length_bytes PASSED                                                 [ 97%]
tests/test_wsgi.py::test_wsgi_multiple_start_response PASSED                                                                    [ 98%]
tests/test_wsgi.py::test_wsgi_multi_body PASSED                                                                                 [100%]

============================================================== FAILURES ===============================================================
_______________________________________ test_thread_critical_Local_remove_all_reference_cycles ________________________________________

    def test_thread_critical_Local_remove_all_reference_cycles() -> None:
        try:
            # given
            # Disable automatic garbage collection and set debugging flag.
>           disable_gc_for_garbage_collection_test()

tests/test_garbage_collection.py:36: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

    def disable_gc_for_garbage_collection_test() -> None:
        # Disable automatic garbage collection. To have control over when
        # garbage collection is performed. This is necessary to ensure that another
        # that thread doesn't accidentally trigger it by simply executing code.
        gc.disable()
    
        # Delete the garbage list(`gc.garbage`) to ensure that other tests don't
        # interfere with this test.
        gc.collect()
    
        # Set the garbage collection debugging flag to store all unreachable
        # objects in `gc.garbage`. This is necessary to ensure that the
        # garbage list is empty after execute test code. Otherwise, the test
        # will always pass. The garbage list isn't automatically populated
        # because it costs extra CPU cycles
>       gc.set_debug(gc.DEBUG_SAVEALL)
E       AttributeError: module 'gc' has no attribute 'set_debug'

tests/test_garbage_collection.py:21: AttributeError

During handling of the above exception, another exception occurred:

    def test_thread_critical_Local_remove_all_reference_cycles() -> None:
        try:
            # given
            # Disable automatic garbage collection and set debugging flag.
            disable_gc_for_garbage_collection_test()
    
            # when
            # Create thread critical Local object in sync context.
            try:
                getattr(Local(thread_critical=True), "missing")
            except AttributeError:
                pass
            # Enforce garbage collection to populate the garbage list for inspection.
            gc.collect()
    
            # then
            # Ensure that the garbage list is empty. The garbage list is only valid
            # until the next collection cycle so we can only make assertions about it
            # before re-enabling automatic collection.
            assert gc.garbage == []
        # Restore garbage collection settings to their original state. This should always be run to avoid interfering
        # with other tests to ensure that code should be executed in the `finally' block.
        finally:
>           clean_up_after_garbage_collection_test()

tests/test_garbage_collection.py:55: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

    def clean_up_after_garbage_collection_test() -> None:
        # Clean up the garbage collection settings. Re-enable automatic garbage
        # collection. This step is mandatory to avoid running other tests without
        # automatic garbage collection.
>       gc.set_debug(0)
E       AttributeError: module 'gc' has no attribute 'set_debug'

tests/test_garbage_collection.py:28: AttributeError
======================================================= short test summary info =======================================================
FAILED tests/test_garbage_collection.py::test_thread_critical_Local_remove_all_reference_cycles - AttributeError: module 'gc' has no attribute 'set_debug'
============================================== 1 failed, 80 passed, 1 xfailed in 11.95s ===============================================
Task was destroyed but it is pending!
task: <Task cancelling name='Task-282' coro=<WsgiToAsgi.__call__() running at /tmp/asgiref/asgiref/wsgi.py:21> wait_for=<Future finished result=None>>
Task was destroyed but it is pending!
task: <Task cancelling name='Task-293' coro=<WsgiToAsgi.__call__() running at /tmp/asgiref/asgiref/wsgi.py:21> wait_for=<Future finished result=None>>
Task was destroyed but it is pending!
task: <Task cancelling name='Task-301' coro=<WsgiToAsgi.__call__() running at /tmp/asgiref/asgiref/wsgi.py:21> wait_for=<Future finished result=None>>
Task was destroyed but it is pending!
task: <Task cancelling name='Task-307' coro=<WsgiToAsgi.__call__() running at /tmp/asgiref/asgiref/wsgi.py:21> wait_for=<Future cancelled>>
Task was destroyed but it is pending!
task: <Task cancelling name='Task-313' coro=<WsgiToAsgi.__call__() running at /tmp/asgiref/asgiref/wsgi.py:21> wait_for=<Future cancelled>>
pypy311-test: exit 1 (12.66 seconds) /tmp/asgiref> pytest -v pid=264527
  pypy311-test: FAIL code 1 (23.48=setup[10.82]+cmd[12.66] seconds)
  evaluation failed :( (24.20 seconds)
```

--- a/tests/test_garbage_collection.py
+++ b/tests/test_garbage_collection.py
@@ -1,4 +1,7 @@
 import gc
+import sys
+
+import pytest
 
 from asgiref.local import Local
 
@@ -29,6 +32,9 @@ def clean_up_after_garbage_collection_test() -> None:
     gc.enable()
 
 
+@pytest.mark.skipif(
+    sys.implementation.name == "pypy", reason="Test relies on CPython GC internals"
+)
 def test_thread_critical_Local_remove_all_reference_cycles() -> None:
     try:
         # given

